commit c4b24121b07d67004d06fd2e7b24c232b28e435d
Author: Guillaume Girol <guillaume.girol@cea.fr>
Date:   Tue Jun 29 12:00:00 2021 +0000

    report the correct number of variables in unsat nnf

diff --git a/src/src_sharpSAT/MainSolver/MainSolver.h b/src/src_sharpSAT/MainSolver/MainSolver.h
index 0fdda13..a3eaf15 100644
--- a/src/src_sharpSAT/MainSolver/MainSolver.h
+++ b/src/src_sharpSAT/MainSolver/MainSolver.h
@@ -420,7 +420,7 @@ public:
 		ofstream out(fileName);
 
 		if (falsify) {
-			out << "nnf 1 0 0" << endl;
+			out << "nnf 1 0 " << bdg_var_count << endl;
 			out << "O 0 0" << endl;
 			return;
 		}
